EchoPro/survey.py

```diff
@@ -36,741 +25,457 @@ class Survey:
-                    "The biomass density must be calculated before performing CV analysis on data!"
-                )
-
-        return run_jolly_hampton(self, nr, lat_inpfc, seed, kriged_data)
-
-    def load_transform_mesh(self):
-        """
-
-        :return:
-        """
-
-        self.kriging_mesh = self.get_kriging_mesh()
-
-        self.kriging_mesh.apply_coordinate_transformation(coord_type='transect')
-        self.kriging_mesh.apply_coordinate_transformation(coord_type='mesh')
-
-    def get_kriging_mesh(self) -> KrigingMesh:
-        """
-        Initializes a ``KrigingMesh`` object using
-        parameters obtained from the configuration
-        files.
-
-        Returns
-        -------
-        KrigingMesh
-            An initialized object that contains the mesh
-            data, functions to transform the mesh, and
-            functions to plot the mesh.
+        init_config_path : pathlib.Path
+            A string specifying the path to the initialization YAML file
+        survey_year_config_path : pathlib.Path
+            A string specifying the path to the survey year YAML file
 
         Notes
         -----
-        This function assigns class variables to the returned object.
-        Specifically, the following class variables are created:
-
-        - ``mesh_gdf`` a GeoPandas Dataframe representing the full mesh
-        - ``smoothed_contour_gdf`` a GeoPandas Dataframe representing
-          the smoothed contour (e.g. 200m isobath)
+        This function parses the configuration files and incorporates them into
+        the Survey class object. This initializes the `config` attribute that 
+        becomes available for future reference and functions.
         """
+        ### Validate configuration files
+        # Retreive the module directory to begin mapping the configuration file location
+        current_directory = os.path.dirname(os.path.abspath(__file__))
 
-        return KrigingMesh(self)
-
-    def get_semi_variogram(
-        self,
-        krig_mesh: KrigingMesh = None,
-        params: vario_param_type = {},
-        warning: bool = True,
-    ) -> SemiVariogram:
-        """
-        Initializes a ``SemiVariogram`` object based on the provided
-        ``KrigingMesh`` object, the calculated areal biomass density,
-        and user provided semi-variogram parameters.
+        # Build the full configuration file paths
+        config_files = [init_config_path, survey_year_config_path]
+        config_file_paths = [os.path.join(current_directory, '..', 'config_files', file_name) for file_name in config_files]
 
-        Parameters
-        ----------
-        krig_mesh : KrigingMesh
-            Object representing the Kriging mesh
-        params : dict
-            Semi-variogram specific parameters. Contains the following
-            parameters:
+        # Check files to ensure they exist
+        config_existence = [os.path.exists(file_path) for file_path in config_file_paths]
```

Comment:
Since both `init_config_path` and `survey_year_config_path` are already Path objects, you can use the functionalities of `Path` in place for the `os.path` ops.

Also, I think this loop `data_file_paths = [os.path.join(self.config['data_root_dir'], filename) for filename in parsed_filenames]` can be omitted by checking `(Path(self.config['data_root_dir']) / filename).exists()` in the loop that produces `data_existence`.